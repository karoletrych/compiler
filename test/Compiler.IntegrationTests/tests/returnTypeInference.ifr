module ReturnTypeInference

fun test (p : int)
{
    if(p == 0)
        return 0;
    else if(p == 1)
        return "1";
    else
        return 2.0;
}

fun main
{
    val r0 = test(0);
    val r1 = test(1);
    val r2 = test(2);
    System::Console:.WriteLine(r0.GetType());
    System::Console:.WriteLine(r1.GetType());
    System::Console:.WriteLine(r2.GetType());
}