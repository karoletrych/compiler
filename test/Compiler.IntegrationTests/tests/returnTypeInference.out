System.Int32
System.String
System.Single
